taf/updater/updater_pipeline.py

```diff
@@ -399,14 +401,25 @@ def check_if_local_repositories_clean(self):
                     return UpdateStatus.SUCCESS
                 else:
                     if auth_repo.something_to_commit():
-                        raise RepositoryNotCleanError(auth_repo.name)
+                        if self.force:
+                            taf_logger.info(
+                                f"Cleaning and resetting repository {auth_repo.name} for forced update"
+                            )
+                            auth_repo.clean_and_reset()
+                        else:
+                            raise RepositoryNotCleanError(auth_repo.name)
                     if auth_repo.is_branch_with_unpushed_commits(
                         auth_repo.default_branch
                     ):
-                        raise UnpushedCommitsError(
-                            auth_repo.name,
-                            auth_repo.default_branch,
-                        )
+                        if self.force:
+                            taf_logger.info(
+                                f"Cleaning and resetting repository {auth_repo.name} for forced update"
+                            )
+                            auth_repo.clean_and_reset()
```

Comment:
If there are unpushed commits, with the `--force` flag, the update is successful, but the commit is not removed. The update probably would've failed if there were fetched commits (merge would fail). We should remove these unpushed commits. We have a couple of other cases here (top commits can be different than the last validated commit). We'll need to handle them as well, but I'll update the issue first. Let's focus on these simpler cases for now. The issue here is that `auth_repo.clean_and_reset()` doesn't remove these additional commits. Let's also make sure that we cover this with a test. After the update, in addition to checking if it was successful, check if the top commit of the repository is equal to the last pushed commit